tests/sentry/incidents/test_logic.py

```diff
@@ -1569,6 +1569,151 @@ def test_update_infer_detection_type(self, mock_seer_request):
 
         assert updated_rule.detection_type == AlertRuleDetectionType.STATIC
 
+    @with_feature("organizations:anomaly-detection-alerts")
+    @patch(
+        "sentry.seer.anomaly_detection.store_data.seer_anomaly_detection_connection_pool.urlopen"
+    )
+    def test_update_dynamic_alerts(self, mock_seer_request):
+        mock_seer_request.return_value = HTTPResponse(status=200)
+
+        dynamic_rule = self.create_alert_rule(
+            sensitivity=AlertRuleSensitivity.HIGH,
+            seasonality=AlertRuleSeasonality.AUTO,
+            time_window=60,
+            detection_type=AlertRuleDetectionType.DYNAMIC,
+        )
+        assert mock_seer_request.call_count == 1
+        mock_seer_request.reset_mock()
+        update_alert_rule(dynamic_rule, time_window=30)
+        assert mock_seer_request.call_count == 0
+        mock_seer_request.reset_mock()
+        update_alert_rule(dynamic_rule, name="everything is broken")
+        assert mock_seer_request.call_count == 0
+        mock_seer_request.reset_mock()
+
+        update_alert_rule(
+            dynamic_rule,
+            query="is:unresolved",
+            time_window=30,
+            detection_type=AlertRuleDetectionType.DYNAMIC,
+        )
+        assert mock_seer_request.call_count == 1
+        mock_seer_request.reset_mock()
+
+        update_alert_rule(
+            dynamic_rule,
+            aggregate="count_unique(user)",
+            time_window=30,
+            detection_type=AlertRuleDetectionType.DYNAMIC,
+        )
+        assert mock_seer_request.call_count == 1
+        mock_seer_request.reset_mock()
+
+        static_rule = self.create_alert_rule(time_window=30)
+        update_alert_rule(
+            static_rule,
+            time_window=30,
+            sensitivity=AlertRuleSensitivity.HIGH,
+            seasonality=AlertRuleSeasonality.AUTO,
+            detection_type=AlertRuleDetectionType.DYNAMIC,
+        )
+        assert mock_seer_request.call_count == 1
+        mock_seer_request.reset_mock()
+
+        percent_rule = self.create_alert_rule(
```

Comment:
Can we make this a separate test?
`test_update_alert_rule_from_percent_to_dynamic`?